core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -251,7 +277,19 @@ public ServerBuilder port(InetSocketAddress localAddress, SessionProtocol protoc
      * @see <a href="#no_port_specified">What happens if no HTTP(S) port is specified?</a>
      */
     public ServerBuilder port(ServerPort port) {
-        ports.add(requireNonNull(port, "port"));
+        requireNonNull(port, "port");
+        for (int i = 0; i < ports.size(); i++) {
+            final ServerPort p = ports.get(i);
+            if (p.localAddress().equals(port.localAddress())) {
+                final Set<SessionProtocol> protocols =
+                        new ImmutableSet.Builder<SessionProtocol>().addAll(p.protocols())
```

Comment:
Consider `Sets.immutableEnumSet`

Reply:
Can't find a simple way to create an immutable enum set from 1 enum and multiple enums. Ended up converting in `ServerPort` constructor.